src/main/java/com/commercetools/sync/producttypes/utils/ProductTypeUpdateAttributeDefinitionActionUtils.java

```diff
@@ -158,15 +159,44 @@
                 final AttributeDefinitionDraft matchingNewAttributeDefinitionDraft =
                     newAttributeDefinitionDraftsNameMap.get(oldAttributeDefinitionName);
                 return ofNullable(matchingNewAttributeDefinitionDraft)
-                    .map(attributeDefinitionDraft ->
-                        buildActions(oldAttributeDefinition, attributeDefinitionDraft)
-                    )
+                    .map(attributeDefinitionDraft -> {
+                        // attribute type is required so if null we let commercetools to throw exception
```

Comment:
This comment means that even knowing that the product type is invalid (because the attribute type is required for an attribute) we don't throw any exception. 

We let commercetools to throw the exception. This is a decision we made because the commercetools documentation states that the attribute type is required, and we decided not throw library validations for those cases that are documented in commercetools.

In this case the validation is just for checking if there is attribute type to avoid a null pointer exception in the `haveSameAttributeType`.